components/docker-auth/pkg/auth/authn_handler.go

```diff
@@ -207,112 +210,105 @@ func validateAccessToken(token string, providedUsername string, execId string) b
 	if err != nil {
-		log.Printf("[%s] Error sending the request to the introspection endpoint : %s\n", execId, err)
-		return false
+		return false, fmt.Errorf("error sending the request to the introspection endpoint : %s", err)
 	}
 
 	if res.StatusCode == http.StatusBadRequest {
-		log.Printf("[%s] %d status code returned from IDP probably due to empty token\n", execId,
+		return false, fmt.Errorf("[%s] %d status code returned from IDP probably due to empty token", execId,
 			res.StatusCode)
-		return false
 	} else if res.StatusCode != http.StatusOK {
-		log.Printf("[%s] Error while calling IDP, status code :%d. Exiting without authorization\n", execId,
-			res.StatusCode)
-		return false
+		return false, fmt.Errorf("[%s] Error while calling IDP, status code :%d. Exiting without "+
+			"authorization\n", execId, res.StatusCode)
 	}
 
 	defer res.Body.Close()
 	body, err := ioutil.ReadAll(res.Body)
 	if err != nil {
-		log.Printf("[%s] Error reading the response from introspection endpoint. Returing without "+
-			"authorization : %s\n", execId, err)
-		return false
+		return false, fmt.Errorf("error reading the response from introspection endpoint. Returing without "+
+			"authorization : %s", err)
 	} else {
-		log.Printf("[%s] Response recieved from introspection endpoint : %s\n", execId, body)
+		logger.Debugf("[%s] Response received from introspection endpoint : %s", execId, body)
 	}
 
-	var result map[string]interface{}
-	err = json.Unmarshal([]byte(string(body)), &result)
+	type Response struct {
+		Active   bool   `json:"active"`
+		Username string `json:"username"`
+		Exp      int64  `json:"exp"`
+	}
+	var response Response
+	err = json.Unmarshal(body, &response)
 	if err != nil {
-		log.Printf("[%s] Error un marshalling the json : %s\n", execId, err)
-		return false
+		return false, fmt.Errorf("error unmarshalling the json. This may be due to a invalid token : %s", err)
 	}
-	isActive, ok := (result["active"]).(bool)
-	if !ok {
-		log.Printf("[%s] Error casting active to boolean. This may be due to a invalid token\n", execId)
-		return false
+	logger.Debugf("[%s] Resolved access token validity", execId)
+	isExpired, err := isExpired(response.Exp, logger, execId)
+	if err != nil {
+		return false, err
 	}
-	log.Printf("[%s] Resolved acess token validity\n", execId)
-	isExpired := isExpired(result["exp"], execId)
-	isValidUser := isValidUser(result["username"], providedUsername, execId)
-	return isExpired && isActive && isValidUser
+	isValidUser, err := isValidUser(response.Username, providedUsername, logger, execId)
+	if err != nil {
+		return false, err
+	}
+	return isExpired && response.Active && isValidUser, nil
 }
 
 // resolves the IS host and port from the environment variables.
 // If the environment is not set the port and host will be resolved through the config file.
-func resolveIntrospectionUrl(execId string) (string, bool) {
+func resolveIntrospectionUrl(logger *zap.SugaredLogger, execId string) (string, bool, error) {
```

Comment:
Why we need 3 things returned from here ? Returning URL would be enough. (you may have an error as well)

Reply:
Fixed by #159